frontend/src/components/DataCard/SuggestedLabels.jsx

```diff
@@ -1,15 +1,21 @@
 import React from "react";
+import * as stringSimilarity from "string-similarity";
 
 export default function SuggestedLabels({ card, labels }) {
     if (labels.length <= 5) {
         return null;
     }
 
+    const labelStrings = labels.map(label => `${label.name} (${label.description})`);
+
+    let matches = stringSimilarity.findBestMatch(card.text.text, labelStrings);
+    matches = matches.ratings.sort((rating1, rating2) => parseFloat(rating2.rating) - parseFloat(rating1.rating)).slice(0, 5);
+
     return (
         <div className="suggestions">
             <h4>Suggested Labels</h4>
-            {card.text.similarityPair.slice(0, 5).map((opt, index) => (
-                <div key={index + 1} className="">{index + 1}. {opt.split(':')[0]}</div>
+            {matches.map((match, index) => (
+                <div key={index + 1} className="">{index + 1}. {match.target}</div>
```

Comment:
I seem to recall that we wanted these suggestions to be buttons. Would it be straightforward to have them be clickable as labels?

Reply:
Sure, I think it should, I can also look into this.